### PR TITLE
Display cloudwatch url as output for cortex logs

### PR DESCRIPTION
**Pull request by vishalbollu**
Display cloudwatch url as the output for cortex logs. Add a flag: `--random-pod` to stream logs from a random pod.

### CODE REVIEW

File: pkg/operator/endpoints/logs.go [hunk 1]
```diff
@@ -67,3 +70,57 @@ func ReadLogs(w http.ResponseWriter, r *http.Request) {
...
+		logURL, err := operator.APILogURL(apiResponse[0].Spec)
+		if err != nil {
+			respondError(w, r, err)
+			return
+		}
+		respondJSON(w, r, schema.LogResponse{
+			LogURL: logURL,
+		})
+	default:
+		respondError(w, r, resources.ErrorOperationIsOnlySupportedForKind(*deployedResource, userconfig.RealtimeAPIKind))
```
Comment (RobertLucian) [inline pkg/operator/endpoints/logs.go:124]: Should this be `resources.ErrorOperationIsOnlySupportedForKind(*deployedResource, userconfig.RealtimeAPIKind, userconfig.AsyncAPIKind)`?

File: pkg/operator/operator/workload_logging.go [hunk 2]
```diff
@@ -42,6 +50,173 @@ const (
...
+
+func APILogURL(api spec.API) (string, error) {
+	partition := "aws.amazon"
+	region := config.ClusterConfig.Region
+	if awslib.PartitionFromRegion(region) == "aws-us-gov" {
+		partition = "amazonaws-us-gov"
+	}
+	logGroup := config.ClusterConfig.ClusterName
+
+	secondsAgo := int(time.Now().Unix() - api.LastUpdated)
```
Comment (RobertLucian) [inline pkg/operator/operator/workload_logging.go:140]: Should we be using this var anywhere?

File: pkg/operator/operator/workload_logging.go [hunk 3]
```diff
@@ -42,6 +50,173 @@ const (
...
+}
+
+func BatchJobLogURL(apiName string, jobStatus status.BatchJobStatus) (string, error) {
+	partition := "aws.amazon"
+	region := config.ClusterConfig.Region
+	if awslib.PartitionFromRegion(region) == "aws-us-gov" {
+		partition = "amazonaws-us-gov"
+	}
+	logGroup := config.ClusterConfig.ClusterName
+	debug.Ppg(jobStatus)
```
Comment (RobertLucian) [inline pkg/operator/operator/workload_logging.go:168]: This can be removed

File: python/client/cortex/client.py [hunk 4]
```diff
@@ -260,33 +240,3 @@ def stop_job(self, api_name: str, job_id: str, keep_cache: bool = False):
         ]
 
         run_cli(args)
-
-    @sentry_wrapper
-    def stream_api_logs(
```
Comment (RobertLucian) [inline python/client/cortex/client.py:265]: What about the `stream_api_logs` and `stream_job_logs` from the `tests.py` file? Should we remove those as well, or reimplement these methods for the `--random-pod` case?

**Review by RobertLucian — APPROVED** [review]

File: pkg/operator/operator/workload_logging.go [hunk 5]
```diff
@@ -42,6 +49,168 @@ const (
...
+	}
+	logGroup := config.ClusterConfig.ClusterName
+	if jobStatus.EndTime != nil {
+		endTime := *jobStatus.EndTime
+		endTime = endTime.Add(60 * time.Second)
+		return completedBatchJobLogsURL(completedJobLogURLTemplateArgs{
+			Partition: partition,
+			Region:    region,
+			StartTime: timeString(jobStatus.StartTime),
+			EndTime:   timeString(endTime),
```
Comment (deliahu) [inline pkg/operator/operator/workload_logging.go:199]: we can discuss live, I was just wondering why the start and end times are necessary

File: python/client/cortex/client.py [hunk 6]
```diff
@@ -114,41 +114,21 @@ def deploy_from_file(
...
-            for c in iter(lambda: process.stdout.read(1), ""):
-                sys.stdout.write(c)
-                sys.stdout.flush()
-
         api_name = deploy_result["api"]["spec"]["name"]
         if deploy_result["api"]["spec"]["kind"] != "RealtimeAPI":
             return deploy_result
 
         env = os.environ.copy()
         env["CORTEX_CLI_INVOKER"] = "python"
```
Comment (deliahu) [inline python/client/cortex/client.py:125]: I think these lines can be removed

File: python/client/cortex/client.py [hunk 7]
```diff
@@ -114,41 +114,21 @@ def deploy_from_file(
...
         time.sleep(5)
 
-        def stream_to_stdout(process):
-            for c in iter(lambda: process.stdout.read(1), ""):
-                sys.stdout.write(c)
-                sys.stdout.flush()
-
         api_name = deploy_result["api"]["spec"]["name"]
         if deploy_result["api"]["spec"]["kind"] != "RealtimeAPI":
             return deploy_result
```
Comment (deliahu) [inline python/client/cortex/client.py:122]: Should these lines be above the `time.sleep`?

Comment (deliahu) [reply python/client/cortex/client.py:122]: EDIT: I suppose it's here for the async API? Does the async API have the same statuses as the realtime API (which would meant that `wait` could be used)?

**Review by deliahu — APPROVED** [review]

**Review by deliahu — APPROVED** [review]

**Pull request merged**